hazelcast-spring/src/main/java/com/hazelcast/spring/transaction/HazelcastTransactionManager.java

```diff
@@ -121,6 +121,26 @@ protected void doBegin(Object transaction, TransactionDefinition definition) thr
         }
     }
 
+    private TransactionContext getTransactionContext(TransactionDefinition definition) {
+        TransactionContext transactionContext;
+        if (definition.getTimeout() == TransactionDefinition.TIMEOUT_DEFAULT
+            && this.getDefaultTimeout() == TransactionDefinition.TIMEOUT_DEFAULT) {
+                transactionContext = hazelcastInstance.newTransactionContext();
+        } else {
+            TransactionOptions options = new TransactionOptions();
+            if (definition.getTimeout() != TransactionDefinition.TIMEOUT_DEFAULT) {
+                options.setTimeout(definition.getTimeout(), TimeUnit.SECONDS);
+            } else {
+                options.setTimeout(this.getDefaultTimeout(), TimeUnit.SECONDS);
+            }
+            transactionContext = hazelcastInstance.newTransactionContext(options);
+        }
```

Comment:
Whaty do you think about simplifying nested if-s? 
```suggestion
        TransactionOptions options = TransactionOptions.getDefault();
        if (definition.getTimeout() != TransactionDefinition.TIMEOUT_DEFAULT) {
            options.setTimeout(definition.getTimeout(), TimeUnit.SECONDS);
        } else if (getDefaultTimeout() != TransactionDefinition.TIMEOUT_DEFAULT) {
            options.setTimeout(getDefaultTimeout(), TimeUnit.SECONDS);
        }
        TransactionContext transactionContext = hazelcastInstance.newTransactionContext(options);
```